documentation/API-Testing-Guidelines.md

```diff
@@ -1,40 +1,284 @@
 # API Testing Guidelines
 
 ## Table of Contents
-1. [Introduction](#introduction)
-2. [Test Cases contributions for APIs](#contribution)
-3. [Location of feature file](#location)
-4. [Best practices and recommendations](#recommendations)
-7. [References](#references)
+- [Introduction](#introduction)
+- [Test Cases contributions for APIs](#contribution)
+- [Location of feature file](#location)
+- [Test plan design guidelines](#guidelines)
+  - [Feature structure](#feature-structure)
+  - [Environment variables](#environment-variables)
+  - [Background section](#background)
+  - [Scenario structure](#scenario-structure)
+  - [Request setup](#request-setup)
+  - [Request sending](#request-sending)
+  - [Response validation](#response-validation)
+- [Test plan coverage](#coverage)
+  - [Independent operations](#independent-operations)
+  - [CRUD operations](#crud-operations)
+  - [Notifications](#notifications)
+  - [Error scenarios](#error-scenarios)
+- [References](#references)
 
 ## Introduction <a name="introduction"></a>
 This document captures guidelines for the API testing in CAMARA project. These guidelines are applicable to every API to be worked out under the CAMARA initiative.
 
 ## Test Cases contribution for APIs <a name="contribution"></a>
 Based on the decision taken in the Commonalities working group and as documented in [API-Readiness-Checklist](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Readiness-Checklist.md), at least one Gherkin feature file will be added to the main subproject repo and this will fulfill the minimum criteria of readiness with respect to API test cases and documentation. If subprojects also intend to add test implementations, an aligned single implementation that is agreed among all provider implementors could be added to the main subproject repo. If no alignment is possible, each provider implementer will add the test implementation to their own repos.
```

Comment:
The updated API Readiness Checklist is proposed in Release Msanagement: https://github.com/camaraproject/ReleaseManagement/pull/29

It impacts this section and possibly "Basic API test cases & documentation" and "Enhanced API test cases & documentation"  should be defined here.

Reply:
It's difficult to set the difference between Basic and Enhanced. To me basic tests cases should cover at least that the implementation interface is compliant with the API spec, in terms of syntax and support for all mandatory aspects. Enhanced test plan may validate unusual error scenarios and corner cases. But this is very subtle.